include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -528,6 +512,31 @@ SECTIONS
 	KEEP(*(.gnu.attributes))
 	}
 
+#if defined(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
+#if CONFIG_ARM_NSC_REGION_BASE_ADDRESS != 0
+	#define NSC_ALIGN . = ABSOLUTE(CONFIG_ARM_NSC_REGION_BASE_ADDRESS)
+#elif defined(CONFIG_CPU_HAS_NRF_IDAU)
+	#define NSC_ALIGN . = ALIGN(0x8000) - (1 << LOG2CEIL(__sg_size))
+#else
+	#define NSC_ALIGN . = ALIGN(4)
+#endif
+
+#ifdef CONFIG_CPU_HAS_NRF_IDAU
+	#define NSC_ALIGN_END . = ALIGN(0x8000)
+#else
+	#define NSC_ALIGN_END . = ALIGN(4)
+#endif
+
+SECTION_PROLOGUE(.gnu.sgstubs,,)
+{
+       NSC_ALIGN;
+       __sg_start = .;
```

Comment:
Basically it isn't necessary, and I don't quite know why. It seems the stubs are placed right after the `.gnu.sgstubs` output section, instead of inside it, regardless of whether `*(.gnu*)` is specified.

Reply:
That's what I observed as well, @wentongwu .

Reply:
1. I think we should add some comments at there in case someone has some question. @oyvindronningstad @ioannisg 

2. "respecting their constrains on region alignment"  "The patch forces the region to be placed at the end of the Firmware flash area."     I can understand the region alignment. Also yeah, it really put the section at the end of the flash, but why should we put that region at the end?  I'm sorry I don't have the spec for nRF SoCs.

Reply:
> 1. I think we should add some comments at there in case someone has same question. @oyvindronningstad @ioannisg

Good point, I'll add comments

> 2. "respecting their constrains on region alignment"  "The patch forces the region to be placed at the end of the Firmware flash area."     I can understand the region alignment. Also yeah, it really put the section at the end of the flash, but why should we put that region at the end?  I'm sorry I don't have the spec for nRF SoCs.

The nRF9160 needs NSC regions to be placed at the end of a 32 kB region.

https://infocenter.nordicsemi.com/topic/ps_nrf9160/spu.html?cp=2_0_0_5_14_1_0#flash_nsc

Reply:
Done